### PR TITLE
Remove service protocol version checks in DevTools

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes https://github.com/flutter/devtools/issues/3699

### CODE REVIEW

File: packages/devtools_app/lib/src/http/http_request_data.dart [hunk 1]
```diff
@@ -507,6 +68,11 @@ class DartIOHttpRequestData extends HttpRequestData {
         .then((updated) => _request = updated);
   }
 
+  static List<Cookie> _parseCookies(List<dynamic>? cookies) {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/http/http_request_data.dart:71]: Should this be `List<dynamic>`? Do we not have a static type to use here?

File: packages/devtools_app/lib/src/http/http_request_data.dart [hunk 2]
```diff
@@ -553,18 +117,65 @@ class DartIOHttpRequestData extends HttpRequestData {
     };
   }
 
+  @override
+  String? get contentType {
+    if (responseHeaders == null || responseHeaders![_contentTypeKey] == null) {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/http/http_request_data.dart:122]: Create a local for `responseHeaders` to avoid the extra checks?

File: packages/devtools_app/lib/src/http/http_request_data.dart [hunk 3]
```diff
@@ -553,18 +117,65 @@ class DartIOHttpRequestData extends HttpRequestData {
...
+    return connectionInfo != null ? connectionInfo[_localPortKey] : null;
+  }
+
   /// True if the HTTP request hasn't completed yet, determined by the lack of
   /// an end event.
   @override
   bool get inProgress =>
       _hasError ? !_request.isRequestComplete : !_request.isResponseComplete;
 
   /// All instant events logged to the timeline for this HTTP request.
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/http/http_request_data.dart:177]: This comment still references the timeline, but you've removed that in this PR, no?

File: packages/devtools_app/linux/flutter/generated_plugins.cmake [hunk 4]
```diff
@@ -7,6 +7,9 @@ list(APPEND FLUTTER_PLUGIN_LIST
   url_launcher_linux
 )
```
Comment (bkonyi) [inline packages/devtools_app/linux/flutter/generated_plugins.cmake:9]: Were these changes intentional?

**Review by bkonyi — APPROVED** [review]
Generally LGTM

File: packages/devtools_app/lib/src/http/http_request_data.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/http/http_request_data.dart:71]: changed to `List<String>`

File: packages/devtools_app/lib/src/http/http_request_data.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/http/http_request_data.dart:122]: done

File: packages/devtools_app/linux/flutter/generated_plugins.cmake (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/linux/flutter/generated_plugins.cmake:9]: removed
